llvm/lib/Target/AIE/AIEPostPipeliner.cpp

```diff
@@ -184,13 +186,12 @@ void PostPipeliner::computeLoopCarriedParameters() {
 
   // Propagate Earliest upstream, initialize Latest
   // Unrestricted: last cycle of last stage
-  const int Latest = NCopies * II - 1;
+  const int Latest = -1;
```

Comment:
Nit: Discussed offline: I'd be happy to have that start at `0` and grow, much like the cycles of the `Bot` zone in `MachineScheduler`

Reply:
nit: Since I use Earliest and Latest together to define the interval in which to schedule, the name Latest is a misfit if it runs the other way. It also requires the top-down or bottom up context to apply the proper negations. The code is fully separated from the machine scheduler and the concept fully separated from the picking order.

Reply:
As I said, I won't block the PR for that, I just find the symmetry appealing if we use something like `TopEarliest` that belongs in `[0,Length)`, and `BotLatest` that has the same interval. But that is purely subjective. My only valid argument is that if we go this way, we would mimic the concepts of the `MachineScheduler`, and I think it's important in a large codebase to share concepts.